src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java

```diff
@@ -12,136 +13,136 @@
+        Set<MultipartFile> wasChecked = new HashSet<>();
+
+        //check all uploaded files to see if they should be paired or left single
+        for (int i = 0; i < files.size(); i++) {
+            file1 = files.get(i);
+
+            boolean pair = false;
+            if (!wasChecked.contains(file1)) {
+                for (int j = i + 1; j < files.size() && !pair; j++) {
+                    file2 = files.get(j);
+
+                    if (!wasChecked.contains(file2)) {
+                        MultipartFile[] filePair = null;
+
+                        List<Diff> diffs = diff.diff_main(file1.getOriginalFilename(), file2.getOriginalFilename());
+                        //The size of `diffs` is 4 when only 1 character differs between the two strings
+                        //if the two files should be paired, this would be the list of diffs between the file names:
+                        //		diffs[0] = common prefix
+                        //		diffs[1] = unique character 1 (e.g. "1" or "f")
+                        //		diffs[2] = unique character 2 (e.g. "2" or "r")
+                        //		diffs[3] = common suffix (e.g. ".fastq")
+                        if (diffs.size() == 4) {
+                            String file1ID = diffs.get(1).text;
+                            String file2ID = diffs.get(2).text;
+                            //Sometimes files uploaded get put in a different ordering such that
+                            //the first file is the "reverse" sequence file and the last file is
+                            //the "forward" sequence file. This long condition checks for that
+                            //situation.
+                            if ((Stream.of(forwardMatches).anyMatch(x -> file1ID.contains(x))
+                                    && Stream.of(reverseMatches).anyMatch(x -> file2ID.contains(x)))
+                                    || (Stream.of(reverseMatches).anyMatch(x -> file1ID.contains(x))
+                                    && Stream.of(forwardMatches).anyMatch(x -> file2ID.contains(x)))) {
+                                filePair = new MultipartFile[]{file1, file2};
+                            }
+                        }
+
+                        if (filePair != null) {
+                            pair = true;
+                            //organizedFiles.put(diffs.get(0).text, Arrays.asList(filePair));
+
+                            pairedFiles.put(diffs.get(0).text, Arrays.asList(filePair));
+
+                            logger.trace("Uploaded files [" + filePair[0].getName() + ", " + filePair[1].getName()
+                                    + "] were paired.");
+                            wasChecked.add(file2);
+                        }
+                    }
+                }
+                if (!pair) {
+                    MultipartFile[] singleFile = {file1};
+                    //organizedFiles.put(file1.getOriginalFilename(), Arrays.asList(singleFile));
```

Comment:
Same here.  Delete it please.

Reply:
Updated in ea379e563c249e785d87b7b7906dd0ef1aa37909